doc/agent-workflow.md

```diff
@@ -0,0 +1,68 @@
+# stOLAS — Agent Workflow
+
+This file provides agent workflow description section. 
+---
+
+## Workflow Diagram
+
+```mermaid
+graph LR
+  StartRound
+  CheckAnyWorkRound
+  WaitingRound
+
+  StartRound-->|DONE|CheckAnyWorkRound
+  CheckAnyWorkRound-->|CLAIM_BRIDGED_TOKEN|ClaimBridgedTokensRound
+  CheckAnyWorkRound-->|CALL_REDEEM|RedeemRound
+  CheckAnyWorkRound-->|CLAIM_REWARDS|ClaimRewardTokensRound
+  CheckAnyWorkRound-->|CALL_CHECKPOINTS|CheckpointRound 
+  CheckAnyWorkRound-->|TRIGGER_L2_TO_L1|TriggerL2ToL1BridgeRound
+  CheckAnyWorkRound-->|NO_WORK|WaitingRound
+  
+
+  ClaimRewardTokensRound-->|DONE|CheckAnyWorkRound
+  ClaimBridgedTokensRound-->|DONE|CheckAnyWorkRound
+  TriggerL2ToL1BridgeRound-->|DONE|CheckAnyWorkRound
+  CheckpointRound-->|DONE|CheckAnyWorkRound
+  RedeemRound-->|DONE|CheckAnyWorkRound
+  WaitingRound-->|DONE|CheckAnyWorkRound
+```
+
+---
+
+## Agent Workflow — Step by Step
+
+### Claim Bridged Tokens
+
+Agents are going to interact with relevant L1 bridge contracts in order to finalize fund transfers from L2 to L1.
+Each native bridge is specific to its fund transfer finalization times and routines.
+
+### Redeem Stake Operations
+
+There could be scenarios when **STAKE** / **UNSTAKE** operations are not complete in an automatic way on L2 when triggered on L1.
+For example, OLAS funds arrive across bridge later than the message with the instruction about where funds need to be relayed.
+In this case, the `RequestQueued()` event in each [DefaultStakingProcessorL2](../contracts/l2/bridging/DefaultStakingProcessorL2.sol)
```

Comment:
So we just wait for this event prior to calling anything, rather than waiting for the funds to show positive no?

Reply:
Yes, the engagement is needed only when `RequestQueued()` pops up. In the majority of cases requests are executed right away.